src/iuse_actor.cpp

```diff
@@ -1799,26 +1799,39 @@ void musical_instrument_actor::load( JsonObject &obj )
     fun_bonus = obj.get_int( "fun_bonus", 0 );
     description_frequency = obj.get_int( "description_frequency" );
     descriptions = obj.get_string_array( "descriptions" );
+    NPCdescriptions = obj.get_string_array( "NPCdescriptions" );
 }
 
 long musical_instrument_actor::use( player &p, item &it, bool t, const tripoint& ) const
 {
     if( p.is_underwater() ) {
-        p.add_msg_if_player( m_bad, _("You can't play music underwater") );
+        if ( p.is_player() ) {
+            p.add_msg_if_player(m_bad, _("You can't play music underwater"));
+        } else {
+            p.add_msg_if_player(m_bad, _("%s can't play music underwater"), p.disp_name(false));
+        }
         it.active = false;
         return 0;
     }
 
     // Stop playing a wind instrument when winded or even eventually become winded while playing it?
     // It's impossible to distinguish instruments for now anyways.
     if( p.has_effect( effect_sleep ) || p.has_effect( effect_stunned ) || p.has_effect( effect_asthma ) ) {
-        p.add_msg_if_player( m_bad, _("You stop playing your %s"), it.display_name().c_str() );
+        if ( p.is_player() ) {
+            p.add_msg_if_player( m_bad, _("You stop playing your %s"), it.display_name().c_str() );
+        } else {
+            p.add_msg_if_player( m_bad, _("%s stops playing their %s"), p.disp_name(false), it.display_name().c_str() );
```

Comment:
The function is `add_msg_player_or_npc`. There should be many examples of it being used in the file.